### PR TITLE
handle apple git

### PR DESCRIPTION
**Pull request by kzw**
### Description

This resolves issue chef issue 6350

### Issues Resolved

https://github.com/chef/chef/issues/6350

### Check List

- [x] New functionality includes tests
- [x] All tests pass
- [ ] RELEASE\_NOTES.md, has been updated if required (not required for bugfixes, required for API changes)
- [x] All commits have been signed-off for the Developer Certificate of Origin. See <https://github.com/chef/chef/blob/master/CONTRIBUTING.md#developer-certification-of-origin-dco>

### CODE REVIEW

File: lib/chef/provider/git.rb [hunk 1]
```diff
@@ -103,8 +105,11 @@ def action_sync
         end
       end
 
-      def git_minor_version
-        @git_minor_version ||= Gem::Version.new( git("--version").stdout.split.last )
+      def git_gem_version
+        output = git("--version").stdout
+        match = GIT_VERSION_PATTERN.match(output)
+        raise ArgumentError, "unparsable git version number #{output}" unless match
```
Comment (lamont-granquist) [inline lib/chef/provider/git.rb]: it'd probably make more sense to leave the version `nil` or `false` or something, while outputting a Chef::Log.warn message and then something like:

```ruby
def git_has_single_branch_option?
  !git_gem_version.nil? && git_gem_version >= Gem::Version.new("1.7.10")
end

[...]

  clone_cmd << "--no-single-branch" if new_resource.depth && git_has_single_branch_option?
```

**Comment by thommay** [comment]
Since we received two PRs at pretty much the same time that both fix the problem, I'm going to make an executive decision and say that we take #6359 , since it's slightly better factored. @Happycoil @kzw thank you both for doing this work!

File: lib/chef/provider/git.rb (hunk 1 above)
Comment (adamleff) [reply lib/chef/provider/git.rb]: I'm of the same opinion as @lamont-granquist. I'd love to see us assume that the `git` available to us will do most of what we need it to do, do a best-effort parsing of the version, and just be selective about the `--no-single-branch` flag.

Comment (kzw) [reply lib/chef/provider/git.rb]: Good suggestions implemented.  Thanks!

**Comment by thommay** [comment]
Thanks for this great PR, @kzw

**Pull request merged**